tests/library/inspector/inspectorTest.ts

```diff
@@ -131,21 +131,27 @@ class Recorder {
   }
 
   async waitForHighlight(action: () => Promise<void>): Promise<string> {
-    let callback;
-    const result = new Promise<string>(f => callback = f);
-    const listener = async msg => {
+    // We get the last highlighted selector, because Firefox sometimes issues multiple
+    // focus events.
+    let generatedSelector: string | undefined;
+    let callback: Function | undefined;
+
+    const listener = async (msg: ConsoleMessage) => {
       const prefix = 'Highlight updated for test: ';
       if (msg.text().startsWith(prefix)) {
-        this.page.off('console', listener);
-        callback(msg.text().substr(prefix.length));
+        generatedSelector = msg.text().substr(prefix.length);
+        if (callback) {
+          this.page.off('console', listener);
```

Comment:
the listener should be removed unconditionally here

Reply:
The whole idea of this patch is to not remove the listener while `action()` is in progress.